lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -44,21 +44,29 @@ def fragment_cache(cached_hash, non_cached_hash)
 
         private
 
-        def add_relationships(resource, name, serializers)
-          resource[:relationships] ||= {}
-          resource[:relationships][name] ||= { data: [] }
-          resource[:relationships][name][:data] += serializers.map { |serializer| { type: serializer.json_api_type, id: serializer.id.to_s } }
+        def resource_identifier_type(serializer)
+          if ActiveModel::Serializer.config.jsonapi_resource_type == :singular
+            serializer.object.class.model_name.singular
+          else
+            serializer.object.class.model_name.plural
+          end
         end
 
-        def add_relationship(resource, name, serializer, val=nil)
-          resource[:relationships] ||= {}
-          resource[:relationships][name] = { data: val }
-
-          if serializer && serializer.object
-            resource[:relationships][name][:data] = { type: serializer.json_api_type, id: serializer.id.to_s }
+        def resource_identifier_id(serializer)
+          if serializer.respond_to?('id')
```

Comment:
I also see you some symbols on the other places and here a string, I'd go with a symbol as well